### PR TITLE
Scale multi-width unicodes

### LINKED ISSUE #513: The double byte character calculating is not correct so the diagram's width is not fitting.
incorrect;
https://play.d2lang.com/?layout=dagre&script=SklKtlKoyU3hUlBQVnjc2PW4cf3jpo7HjW1cCgq6Cmn5-WA6KbGIq4YLEAAA__8%3D

adding single byte charactors(\x20);
https://play.d2lang.com/?layout=dagre&script=SklKtlKoyU3hUlBQVnjc2PW4cf3jpg4FKHjc2MaloKCrkJafD6aTEou4argAAQAA__8%3D

There are four double byte charactors.
I need to add four single byte charactors.

### LINKED ISSUE #749: Markdown blocks have trouble rendering unicode emoji
#### Summary

When using unicode emoji in a markdown block, the output appears to be getting trimmed. I tried to see if this was an issue with [goldmark](https://github.com/yuin/goldmark), but I don't think it is. I see that there is [goldmark-emoji](https://github.com/yuin/goldmark-emoji) for `:joy:`-style emojis, but it didn't look like d2 was using this.

Issue appears to be for all layout engines.

Open to working on this if provided some guidance (would be my foray into the codebase).

#### Resources

<details>
<summary>D2 test file</summary>

```d2
Single emoji: |md
  🐵
|

Single emoji with text at beginning: |md
  This is a monkey emoji: 🐵
|

Single emoji with text at end: |md
  🐵 That was a monkey emoji.
|

Single emoji with text around: |md
  This is a monkey emoji: 🐵 It's pretty cool.
|

Multiple emojis: |md
  🐵🐵🐵🐵
|

Multiple emojis with text at beginning: |md
  These are monkey emojis: 🐵🐵🐵🐵
|

Multiple emojis with text at end: |md
  🐵🐵🐵🐵 Those were monkey emojis.
|

Multiple emojis with text around: |md
  These are monkey emojis: 🐵🐵🐵🐵 They are pretty cool.
|
```
</details>

&mdash; [Playground Link](https://play.d2lang.com/?script=lJHBjcIwEEXvruLf9rYFpIM97Ik0YMgoHkhslBkUIqUIOqAK-qIERCDCWFEChzn5-83384p9WRGoDlvO0NeFAa7n08X0xsRnaFkdlI4Kq1hTyd6zL8cruWMBCyzq4HfUjcBlFPki3ovcWUVrU9TvHKUJhxdlrgr-9Eewb0i1wyaEasD-Pz_0kZS4TjwT0UUrJATb0FsVyb5HJ5biQe6CEFpK10y9bcbah1XvyW4IJhpvAQAA__8%3D&layout=dagre&theme=0&sketch=0&)

<details>
<summary>Exported SVG & PNG</summary>

Exported SVG
Exported PNG
</details>

### LINKED ISSUE #813: container title
v0.1.6 -> no problem
v0.2.0 -> degrade 😢

https://play.d2lang.com/?l=&script=HMq7CYAwFEDRPlvfxyOIneCHLKBi4wY6zF1ETHngyCi3MUgtPyKlySuXLJJFQiZ5ZJO9J-SQU6rMskr20MoXAAD__w%3D%3D&

### PR DESCRIPTION
**Pull request by alixander**
screencapture-file-Users-alexanderwang-dev-alixander-d2-e2etests-out-e2e-report-html-2023-02-14-10_15_07

closes #813 
closes #513 
closes #749 
probably missing like 5 other issues @bo-ku-ra 's opened ;)

### CODE REVIEW

**Comment by bo-ku-ra** [comment]
awesome!

Emoji ZWJ Sequence / U+200D ZERO WIDTH JOINER (ZWJ) 
https://emojipedia.org/emoji-zwj-sequence/

**Comment by bo-ku-ra** [comment]
i think the test would require a mixed text of double-byte charactors, single-byte charactors, and emojis, too.

**Comment by bo-ku-ra** [comment]
{style.stroke-width: 15; style.double-border: true}
- width 15:max + double-border(v 0.2.0)

**Comment by bo-ku-ra** [comment]
{style.font-size:100}

https://play.d2lang.com/?l=&script=SrRSeNzY9bip5XHjzMeN-x83rnrcOPVxY4tCNdhwPXiAWSkYGhhYK0BEkV1ipWBoChNHiSQrhZKi0tRariQCVpBkWDLx7rUgy7WJCrp2Ckm6dslcgAAAAP__&

**Comment by alixander** [comment]
@bo-ku-ra updated with your tests. the method is not 100% accurate, it gives it a little too wide, but I think by an acceptable amount. What do you think

**Comment by alixander** [comment]
@nhooyr tagged you to just take a cursory glance in case anything in the PR raises a flag, no need to dig into

File: ci/release/changelogs/next.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #### Features 🚀
 
+- Multi-byte characters (e.g. Chinese, Japanese, Korean, emojis) are supported. [#817](https://github.com/terrastruct/d2/pull/817)
```
Comment (nhooyr) [inline ci/release/changelogs/next.md:3]: Could you clarify here for users that they were supported before but now are measured correctly.

**Review by nhooyr — APPROVED** [review]
✅

**Review by gavin-ts — APPROVED** [review]

**Comment by bo-ku-ra** [comment]
@alixander
d2 v0.2.0  was designed with a shorter vertical length, but it looks good.
 ( maybe the whole diagram looks squashed? )

**Comment by bo-ku-ra** [comment]
fyi
https://en.wikipedia.org/wiki/Golden_ratio
https://en.wikipedia.org/wiki/Silver_ratio  (<- japanese like it. mickey mouse, doraemon, google logo, etc.)

**Pull request merged**